### ISSUE TITLE
Plugin thinks type is imported when it is not

### ISSUE THREAD

**Issue by TobiasDeBruijn** [intellij-rust/intellij-rust#9720]
## Environment

* **IntelliJ Rust plugin version:** 0.4.182.4996-222
* **Rust toolchain version:** 1.62.0-beta.7 (747075d9c 2022-06-26) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.2.4 (CL-222.4345.21)
* **Operating system:** Linux 5.15.0-52-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc

## Problem description
The IDE seems to think sometimes that a type is imported (thus showing up white, rather than red), hitting Ctrl+space does not do anything. The type is not imported though (and rustc confirms)

## Steps to reproduce
Honestly, I dont know for sure. It usually works, except for one type in my project. (HttpError, an enum, in my case)

I unfortunately cannot include the codebase publicly, though I can share it privately with this project's maintainers, if need be.

**Comment by neonaot** [comment]
Hi!
To see problems, you can disable the option `Ignore unresolved references without quick fix` in  `Settings | Editor | Inspections | Rust | Unresolved reference`. It can produce some false positive errors (for example, if the item is generated by macros, that plugin didn't expand), so it is configured this way by default. 
<img width="261" alt="image" src="https://user-images.githubusercontent.com/49211026/201543411-f12df7f0-0342-4892-8816-2221b80a0d7f.png">

But if the plugin doesn't suggest importing items in a resolvable case, then this is a problem and a code example that reproduces it would be useful.

**Comment by TobiasDeBruijn** [comment]
I think it's the second case: The plugin doesn't resolve an item that is resolvable. The item I mentioned in my original post, `HttpError` is in a super module from where it should be imported. Its an enum with quite some variants, the `Debug` (from `std`) and `Error` (from `thiserror`) derive macros are applied to it. Macro expansion is turned on.

I'd love to share a code example, but I'm unsure *what* causes the issue. The `HttpError` type is visible from the module where it is needed, and autocomplete works. I'm thinking it might be a `pub use` somewhere that causes the plugin to be confused, but that'll take some digging to find, though usually the IDE will then give me multiple import options (and show the item in red), rather than nothing at all.

**Comment by dima74** [comment]
> I unfortunately cannot include the codebase publicly, though I can share it privately with this project's maintainers, if need be.

Could you share the code through https://uploads.jetbrains.com/?

**Comment by TobiasDeBruijn** [comment]
Sure! Upload id: 2022_11_16_7ka7RD9VcHcmAnMLPu4R3k (file: repo.tar.gz)

The HttpError type is defined in `smartbathroom-server/src/error.rs`, the file where I wanted to use it is at `smartbathroom-server/src/routes/v1/data/organizations/locations/rooms/controllers/sensors/alerts/list.rs`

**Comment by dima74** [comment]
@TobiasDeBruijn thanks!

Minimum reproduction:

```rust
mod inner {
    pub fn func() {}
}

macro_rules! gen {
    () => {
        {
            use inner::func;
        }
    };
}

fn main() {
    gen!();
    func();
} //^
```
